### PR TITLE
enhancement: add outdated node url state for removed proposals

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/OutdatedNodeTooltip.svelte [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+</script>
+
+<Tooltip {anchor} {position}>
+    <div class="flex flex-col text-left">
+        <Text type={TextType.p} fontWeight={FontWeight.semibold} fontSize="16" classes="mb-2">
+            {localize('tooltips.outdatedNode.title')}
+        </Text>
+        <Text type={TextType.p} fontWeight={FontWeight.normal}>
+            {localize('tooltips.outdatedNode.body')}
+        </Text>
```
Comment (MarkNerdi) [inline packages/shared/components/OutdatedNodeTooltip.svelte:17]: `p` is the default value for `Text`s

```suggestion
        <Text fontWeight={FontWeight.semibold} fontSize="16" classes="mb-2">
            {localize('tooltips.outdatedNode.title')}
        </Text>
        <Text fontWeight={FontWeight.normal}>
            {localize('tooltips.outdatedNode.body')}
        </Text>
```

File: packages/shared/lib/contexts/governance/actions/getAccountsParticipationEventStatusForEvent.ts [hunk 2]
```diff
@@ -11,6 +16,18 @@ export async function getAccountsParticipationEventStatusForEvent(
...
+        /* eslint-disable no-console */
+        console.error(err)
+        const isEventError = err?.error?.match(/(the requested data)|(was not found)/)?.length > 0
+        if (isEventError) {
+            const proposal = get(registeredProposalsForSelectedAccount)[eventId]
+            if (proposal.status !== ProposalStatus.Ended) {
+                addOrUpdateProposalToRegisteredProposals(
+                    { ...proposal, isNodeOutdated: true },
+                    get(selectedAccountIndex)
+                )
```
Comment (MarkNerdi) [inline packages/shared/lib/contexts/governance/actions/getAccountsParticipationEventStatusForEvent.ts:28]:
```suggestion
                addOrUpdateProposalToRegisteredProposals(
                    { ...proposal, isNodeOutdated: true },
                    account.index
                )
```

File: packages/shared/components/atoms/pills/ProposalStatusPill.svelte [hunk 3]
```diff
@@ -2,8 +2,11 @@
     import { Pill } from 'shared/components'
     import { localize } from '@core/i18n'
     import { ProposalStatus } from '@contexts/governance/enums'
+    import Icon from '../Icon.svelte'
 
     export let status: string
+    export let isNodeOutdated: boolean
+    export let icon: Icon
```
Comment (MarkNerdi) [inline packages/shared/components/atoms/pills/ProposalStatusPill.svelte:9]: this type should not be `Icon` from `Icon.svelte`  but from the `icon.enum`.

File: packages/desktop/views/dashboard/governance/views/ProposalDetailsView.svelte [hunk 4]
```diff
@@ -202,7 +199,7 @@
     }
 
     onMount(async () => {
-        pollParticipationEventStatus($selectedProposal?.id).then(() => (statusLoaded = true))
+        pollProposalData($selectedProposal?.id).then(() => (statusLoaded = true))
```
Comment (MarkNerdi) [inline packages/desktop/views/dashboard/governance/views/ProposalDetailsView.svelte:202]: because we do this fetch only here in the details view, the pill in the proposalslist only updates after we went to the details once.
Maybe we should have 1 `await getParticipationEventStatusForEvent(eventId)` for each event one time on load.

**Review by MarkNerdi — CHANGES_REQUESTED** [review]
UI looks really nice :fire:
I commented a  few minor improvements

File: packages/shared/lib/contexts/governance/actions/addProposalsFromParticipationEventMap.ts [hunk 5]
```diff
@@ -8,6 +8,6 @@ import { createProposalFromEvent } from '../utils'
 export function addProposalsFromParticipationEventMap(eventMap: ParticipationEventMap, account: IAccountState): void {
     Object.values(eventMap).forEach((event: ParticipationEventWithNodes) => {
         const proposal = createProposalFromEvent(event)
-        addOrUpdateProposalToRegisteredProposals(proposal, account.index)
+        addOrUpdateProposalToRegisteredProposals({ ...proposal }, account.index)
```
Comment (MarkNerdi) [inline packages/shared/lib/contexts/governance/actions/addProposalsFromParticipationEventMap.ts:11]: why is this necessary?

File: packages/shared/lib/contexts/governance/utils/createProposalFromError.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+import { IAccount } from '@core/account/interfaces'
+import { networkStatus } from '@core/network/stores'
+
+import { ProposalErrorMode, ProposalStatus } from '../enums'
+import { IProposal } from '../interfaces'
+import { getProposalStatusForMilestone } from './getProposalStatusForMilestone'
+
+export function createProposalFromError(
+    proposal: IProposal,
```
Comment (MarkNerdi) [inline packages/shared/lib/contexts/governance/utils/createProposalFromError.ts:11]: change this to `IProposalMetadata` because the status is calculated in this function

File: packages/shared/lib/contexts/governance/actions/initializeRegisteredProposals.ts [hunk 7]
```diff
@@ -28,7 +31,16 @@ async function getParticipationEventsAndCreateProposalsForAccount(
...
     for (const event of Object.values(events)) {
-        proposals[event.id] = createProposalFromEvent(event)
+        const proposal = createProposalFromEvent(event)
+        try {
+            await getAccountsParticipationEventStatusForEvent(event.id, account)
+            proposals[event.id] = proposal
+        } catch (err) {
+            proposals[event.id] = {
+                ...proposal,
+                ...createProposalFromError(<IProposal>proposal, account, err),
```
Comment (MarkNerdi) [inline packages/shared/lib/contexts/governance/actions/initializeRegisteredProposals.ts:41]: remove this type cast

File: packages/shared/lib/contexts/governance/actions/getAccountsParticipationEventStatusForEvent.ts [hunk 8]
```diff
@@ -1,16 +1,13 @@
...
     eventId: ParticipationEventId,
     account: IAccount = get(selectedAccount)
 ): Promise<ParticipationEventStatus> {
-    try {
-        const status = await account?.getParticipationEventStatus(eventId)
-        return status
-    } catch (err) {
-        return undefined
-    }
+    return account?.getParticipationEventStatus(eventId)
```
Comment (MarkNerdi) [inline packages/shared/lib/contexts/governance/actions/getAccountsParticipationEventStatusForEvent.ts:12]: because of this change, you need to add a try catch in `selected-participation-event-status.store`. 
If not, the questions load permanently on detail view

File: packages/shared/lib/contexts/governance/interfaces/proposal-state.interface.ts [hunk 9]
```diff
@@ -25,4 +25,5 @@ export interface IProposalMetadata {
     additionalInfo: string
     title: string
     nodeUrl: string
+    errorMode?: ProposalErrorMode
```
Comment (MarkNerdi) [inline packages/shared/lib/contexts/governance/interfaces/proposal-state.interface.ts:28]: why `errorMode`? i think naming it `error: ProposalError` is better

File: packages/shared/lib/contexts/governance/utils/createProposalFromError.ts (hunk 6 above)
Comment (Tuditi) [reply packages/shared/lib/contexts/governance/utils/createProposalFromError.ts:11]: Also `account` doesn't seem to be used.

**Review by Tuditi — COMMENTED** [review]
Additionally, I think it should be possible to remove the proposal if the node url is outdated.

**Review by Tuditi — APPROVED** [review]
LGTM!
